### ISSUE TITLE
Reproduce result with TOX-171 and PBMC

### ISSUE THREAD

**Issue by orItach** [jcyang34/lspin#3]
Hi!
The result are fantastic, and the whole idea of two network is really refreshing.
I'm trying to reproduce result on the biological datasets: BASEHOCK, COLON, TOX-171 and PBMC.
I'm succesfully reproduce the result with BASEHOCK and COLON datasets,
thanks you very much for the hyper-parameters.
The hyper-parameters for COLON and BASEHOCK work perfect!
Unfortunately even with the hyper-parameters i'm couldn't reproduce the TOX-171 and PBMC results.
At the hyper-parameters for TOX-171 it's give result of 0.5,
after some tunning the hyper-parameters it's seems need 20,000 epoch
(about 3 times the 7000 you write at the hyper-parameters) to get good result, about 0.8.
Can you double check the hyper-parameters?
Please see the attachment result and the example code.
result:
![image](https://user-images.githubusercontent.com/24195443/219966910-176057ee-7ba1-42bc-af79-5e548ba652ba.png)
[TOX-171.txt](https://github.com/jcyang34/lspin/files/10777623/TOX-171.txt)
The only explanation i'm thinking about it is the encode to one hot vector, i'm couldn't use your function because it's result with array of five slots instead of four, the number of class.

At PBMC i'm also couldn't reproduce the result, probably because i'm made a mistake in the preprocessing you have done and describe at section B.5.1 can you share with use the code of this preprocessing steps?
Thanks in advance

**Comment by orItach** [comment]
Hi again,
I would greatly appreciate some help, currently i stuck because i want your result will be the ground truth of my resarch.
so everything will be helpfull.
Thanks you very much

**Comment by jcyang34** [comment]
Sorry I was busy the last few days. Looking at your code it seems you didn't standardize the data. I think applying z-scoring to the features should solve the problem.

**Comment by orItach** [comment]
Hi, thanks you very much, it's was the problem.
Now the result are almost perfect, between 0.9411 to 1.0.
Can you share with us the code of preprocessing you done on the PBMC dataset?
I'm probably stuck on very little things.

**Comment by jcyang34** [comment]
Hi, the processing codes were a set of R scripts and I reorganized it in preprocessing.R under PBMC_processed/ . I also uploaded the post-processing data called PBMC_splited.npy, which is split into validation set and remaining data (training and testing). Hope this helps.
